### DISCUSSION TITLE
Limiting users for a project?

### DISCUSSION THREAD

**Discussion by benr77** [zitadel/zitadel#8397] [Q&A] [answered]
My understanding is that within an Organisation I have multiple Projects and multiple Users - but any of those users can login to any of the projects within the same Organisation. Is this correct?

I have a use-case whereby we need only selected users to be able to login (for admins of our app) - currently the app just checks if the user can login to the project, and does not check e.g. assigned roles. 

I can of course create a second project (for non-admin users to login) but we need a way to restrict which users are in which project etc. Apparently this is a standard feature on other IAM providers such as Keycloak or Google OIDC, but I'm not sure.

Is this possible? Is the only way to get the app itself to check a role has been assigned in Zitadel? Any pointers appreciated. Thanks.

**Comment by hifabienne** [comment, marked as answer]
Hi @benr77 

So in ZITADEL we do differ between authentication (login) and authorization (permission).
Per default all users in your ZITADEL instance across all organizations are able to login to any project.
If you do want to restrict that, you can enable some settings on the project.
1. Check for Project on Authentication:
This means, the organization of the user who wants to authenticate, needs to have the project. Either as owner or as granted project
<img width="790" alt="image" src="https://github.com/user-attachments/assets/867e9b1f-3a21-4edf-9679-1fd4447fa955">

2. Check authorization on Authentication:
This means the user needs an authorization for that project. All users which do not have an authorization will not be allowed to login
<img width="790" alt="image" src="https://github.com/user-attachments/assets/65f0d0bc-efea-40ac-b41e-4d646d7f5bcd">

3. If you do want to check for specific roles, you can add some ZITADEL actions (javascrip code), which will check if the user does have the permission

Generally we always recommend to also check the permission in your application.

I hope this makes it a little clearer, let me know if you have more questions.

**Reply by benr77** [reply]
Thank you for your comprehensive answer.
